forum/es_helper.py

```diff
@@ -0,0 +1,457 @@
+            query = {"updated_at": {"$gte": start_time}}
+            for response in self._import_to_es(model, index_name, batch_size, query):
+                self.batch_import_post_process(response, current_batch)
+                current_batch += 1
+        log.info(f"Catch up from {start_time} complete.")
+
+    def create_indices(self) -> list[str]:
+        """
+        Create new indices for each model with the current timestamp in the index name.
+
+        Returns:
+            list[str]: List of newly created index names.
+        """
+        index_names = []
+        time_now = datetime.now().strftime("%Y%m%d%H%M%S%f")
+
+        for model in self.models:
+            index_name = f"{model.index_name}_{time_now}"
+            index_names.append(index_name)
+            self.client.indices.create(
+                index=index_name, body={"mappings": model.mapping()}
+            )
+        log.info(f"New indices {index_names} are created.")
+        return index_names
+
+    def delete_index(self, name: str) -> None:
+        """
+        Delete the specified index.
+
+        Args:
+            name (str): The name of the index to delete.
+        """
+        self.client.indices.delete(index=name)
+        log.info(f"Deleted index: {name}.")
+
+    def delete_indices(self) -> None:
+        """
+        Delete all temporary indices.
+        """
+        if self.temporary_index_names:
+            try:
+                self.client.indices.delete(index=self.temporary_index_names)
+                log.info(f"Indices {self.temporary_index_names} are deleted.")
+            except exceptions.NotFoundError as error:
+                log.error(
+                    f"Enable to delete indices {self.temporary_index_names}: {error}"
+                )
+        else:
+            log.info("No Indices to delete.")
+
+    def batch_import_post_process(
+        self, response: tuple[int, Any], batch_number: int
+    ) -> None:
+        """
+        Process the response from a batch import operation.
+
+        Args:
+            response (tuple[int, list[dict]]): Tuple containing the count of successful imports and a list of errors.
+            batch_number (int): The current batch number being processed.
+        """
+        success_count, errors = response
+        for item in errors:
+            if "error" in item["index"]:
+                log.error(f"Error indexing. Response was: {response}")
+        log.info(
+            f"Imported {success_count} documents to the batch {batch_number} into the index"
+        )
+
+    def get_index_shard_count(self, name: str) -> int:
+        """
+        Retrieve the shard count for the specified index.
+
+        Args:
+            name (str): The name of the index.
+
+        Returns:
+            int: Number of shards for the index.
+        """
+        es_settings = self.client.indices.get_settings(index=name)
+        return int(es_settings[name]["settings"]["index"]["number_of_shards"])
```

Comment:
Can it be done like this to avoid breaking or something else.
```
some_default_value = 100
number_of_shards = int(
    es_settings.get(name, {})
    .get("settings", {})
    .get("index", {})
    .get("number_of_shards", some_default_value)
)
return number_of_shards
```

Reply:
Instead of receiving the default value or something, I believe it's better to leave it as is. This will allow us to capture errors in detail since these jobs are not at the user end.